libbeat/kibana/index_test.go

```diff
@@ -0,0 +1,174 @@
+package kibana
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInit(t *testing.T) {
+	beatDir := tmpPath()
+	defer teardown(beatDir)
+
+	//requires all args
+	tests := []struct {
+		Index Index
+	}{
+		{Index: Index{IndexName: "beat-index", BeatDir: beatDir, BeatName: "mybeat."}},
+		{Index: Index{Version: "6.0", BeatDir: beatDir, BeatName: "mybeat."}},
+		{Index: Index{Version: "6.0", IndexName: "beat-index", BeatName: "mybeat."}},
+		{Index: Index{Version: "6.0", IndexName: "beat-index", BeatDir: beatDir}},
+	}
+	for idx, test := range tests {
+		err := test.Index.init()
+		msg := fmt.Sprintf("(%v): Should have raised error", idx)
+		assert.Error(t, err, msg)
+	}
+
+	//checks for fields.yml
```

Comment:
Should we move these tests to a separate test to not mix with the table tests above?